tests/promise/promise.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* globals describe, it, expect, CKEDITOR, window */
+
+'use strict';
+
+describe( 'Promise', function() {
```

Comment:
Well, the test is doing the most basic thing: it creates a Promise and consumes it. I've made it async just to make it more realistic and the setTimeout inside then() is mandatory.

As for "CKEDITOR.require call inside the test function", this is a broader topic that doesn't touch this PR. I've been talking with @gregpabian and we'll be investigating how to change this because this is touching every single test. Still it should not be a blocker here.

Then, when it comes to the returning Promises to Mocha, this is to consider and it requires [Chai as Promised](https://github.com/domenic/chai-as-promised/). I talked with @gregpabian about this as well. Still here we want to test the basics of Promise so this kind of magix is out of scope.